tf-deployer/vsphere_named_machines.yaml

```diff
@@ -91,189 +91,145 @@ items:
 
       disk {
         label            = "${var.disk_label}"
-        size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
+        size             = "${max(var.disk_size, data.vsphere_virtual_machine.template.disks.0.size)}"
         eagerly_scrub    = "${data.vsphere_virtual_machine.template.disks.0.eagerly_scrub}"
         thin_provisioned = "${data.vsphere_virtual_machine.template.disks.0.thin_provisioned}"
       }
 
       clone {
         template_uuid = "${data.vsphere_virtual_machine.template.id}"
-
-        customize {
-          linux_options {
-            host_name = "${var.vm_name}"
-            domain    = "${var.virtual_machine_domain}"
-          }
-
-          network_interface {
-            ipv4_address = "${cidrhost(var.virtual_machine_network_address, var.virtual_machine_ip_address_start + count.index)}"
-            ipv4_netmask = "${element(split("/", var.virtual_machine_network_address), 1)}"
-          }
-
-          ipv4_gateway    = "${var.virtual_machine_gateway}"
-          dns_suffix_list = ["${var.virtual_machine_domain}"]
-          dns_server_list = ["${var.virtual_machine_dns_servers}"]
-        }
       }
 
-      provisioner "file" {
-        source      = "/tmp/machine-startup.sh"
-        destination = "/tmp/master.sh"
-
-        connection {
-          type        = "ssh"
-          private_key = "${file("~/.ssh/vsphere_tmp")}"
-          user        = "ubuntu"
-          agent       = true
-        }
+      cdrom {
+        client_device = true
       }
 
-      // Copy the private key over so the controller is able to ssh into the nodes.
-      provisioner "file" {
-        source      = "~/.ssh/vsphere_tmp"
-        destination = "~/.ssh/id_rsa"
-
-        connection {
-          type        = "ssh"
-          private_key = "${file("~/.ssh/vsphere_tmp")}"
-          user        = "ubuntu"
-          agent       = true
+      vapp {
```

Comment:
Can you please add a comment here about what's going on in this block? Specifically with `user-data`.